pkg/kubelet/kubelet.go

```diff
@@ -1266,6 +1266,67 @@ func (kl *Kubelet) setupDataDirs() error {
 	return nil
 }
 
+// chownDirForRemappedIDs change dir and path ownerships if UserNamespace is enabled at container runtime
+// noop if UserNamespace is disabled
+func (kl *Kubelet) chownDirForRemappedIDs(path string) error {
+	isUseNamespaceSupportedAndEnabled, err := kl.isUserNamespaceRemappingEnabledAtRuntime()
+	if err != nil {
+		klog.Warningf("error while determining usernamespace configuration at runtime: %v", err)
+		return nil
+	}
+	if !isUseNamespaceSupportedAndEnabled {
+		// No-Op if UserNamespace remapping is not enabled at runtime
+		return nil
+	}
+	if path == "" {
+		return fmt.Errorf("path to be setup is empty")
+	}
+	if err := kl.chownAllFilesAt(path); err != nil {
+		return fmt.Errorf("error setting ownership to remapped IDs on %s: %v", path, err)
+	}
+	return nil
+}
+
+// chownAllFilesAt traverses the directory tree at the give path and chowns the paths to adjust for the remapped usernamespaces
+func (kl *Kubelet) chownAllFilesAt(dir string) error {
+	var files []string
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		files = append(files, path)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	klog.V(5).Infof("Chowned paths %v", files)
+	for _, file := range files {
+		_, fileGID, err := kl.getOwnerIDsFor(file)
+		if err != nil {
+			return fmt.Errorf("error in getting volume path owner UID/GID: %v", err)
+		}
+		if fileGID != 0 {
+			klog.V(5).Infof("GID, %v, for path %v is not equal to 0. Skipping chowing assuming it to be FsGroup GID ", fileGID, file)
+			continue
```

Comment:
I think this `continue` is not what we want: If `/var/lib/kubelet/pods` is owned by `200000` (because of your previous test with docker and `/etc/subuid` configured with `200000`), then this code skips the `chown` when you give `containerd/cri` a try with the `100000` configuration, even though the owner is not correct.